### ISSUE TITLE
Possible missing setup.py for DP

### ISSUE THREAD

**Issue by uri200** [magma/magma#11699]
**Problem:**
Domain Proxy is not sharing dependencies with orc8r and AGW. That may cause issues with different versions of same libraries on different gateways. Note we already experienced one issue when AGW was using protobuf 3.14 while the code requiered 3.17

**How It works now:**
In order to build AGW and Orc8 we have some automation to generate the list of python modules and its versions that are proper for our code across all gateways.

To do so we execute [pydep](http://github.com/magma/magma/blob/72f288ec317d5c8fb93c43edaa3deb6a954d5d1e/lte/gateway/release/pydep#L1-L1), for example in [here](http://github.com/magma/magma/blob/72f288ec317d5c8fb93c43edaa3deb6a954d5d1e/lte/gateway/fabfile.py#L114-L114). That will generate a file with a list of dependencies to be installed.

**Possible solution**:
To make sure that dependencies needed by DP are in sync with AGW and Orc8r and nothing breaks you may need to generate a `setup.py` for DP.

You can add then that new setup.py to the fab file that builds magma [here](http://github.com/magma/magma/blob/72f288ec317d5c8fb93c43edaa3deb6a954d5d1e/lte/gateway/fabfile.py#L114-L114)

Examples of setup.py for AGW and Orc8r
http://github.com/magma/magma/blob/72f288ec317d5c8fb93c43edaa3deb6a954d5d1e/orc8r/gateway/python/setup.py#L23-L23

http://github.com/magma/magma/blob/72f288ec317d5c8fb93c43edaa3deb6a954d5d1e/lte/gateway/python/setup.py#L23-L23

**Comment by ardzoht** [comment]
As a workaround we ended up upgrading the protobuf version on `orc8r`: https://github.com/magma/magma/pull/11698/, but we may need to revise it for the `dp` module

**Comment by jpad-freedomfi** [comment]
@uri200 and @ardzoht ack. 

@xbend can you guys take a look at this?

**Comment by xbend** [comment]
@uri200 @ardzoht @jpad-freedomfi 
Domain Proxy has no python packages built for AGW. Domain Proxy is containerized and lives only in k8s space of the orc8r. 
The only thing that AGW shares with Domain Proxy are the protobuf definitions for gRPC APIs.

I think I dont understand the issue. I see two items:
* The issue title mentions missing setup.py, but setup.py is typically used for building and distributing python packages. There is no `dp` python package built at all.
* The issue description mentions version differences of toolsets. Domain Proxy recently converted its protoc compiler version to use 3.10 (to be alligned with orc8r code generation toolset and CI) (https://github.com/magma/magma/pull/11431).

**Comment by uri200** [comment]
Hey @xbend in the past we used to run only `setup.py` from AGW when building AGW. But eventually that broke due to increased shared code between AGW and Orc8r. We ended up having to run both AGW and Orc8r `setup py` when building the gateways. The reason was that not running them meant the gateways ended up with different versions of some libraries that may not be compatible.

Note that even the gateway are separated in folders, they share many files. Also protos are a conflict point if they are built with different versions.  The PR attached to this issue may not be related to DP. It is just an example how a non tracked dependency caused AGW not to run properly due  AGW having installed an older protobuf library version than what the compiled files required to. 

So what it is described here is a potential issue (a change on DP that may cause AGW not to produce a proper install package) that may happen in the future.
